src/sardana/macroserver/msmacromanager.py

```diff
@@ -1290,9 +1294,31 @@ def __runXMLMacro(self, xml):
     _runXMLMacro = __runXMLMacro
 
     def runMacro(self, macro_obj):
+
         name = macro_obj._getName()
         desc = macro_obj._getDescription()
         door = self.door
+        self.macro_obj = macro_obj
+
+        if macro_obj.getParentMacro() is None:
+            try:
+                self.logging_onoff = macro_obj.getEnv("LogMacroOnOff")
+            except:
+                self.logging_onoff = False
+            if self.logging_onoff:
+                try:
+                    self.logging_path = macro_obj.getEnv("LogMacroPath")
+                except:
+                    macro_obj.setEnv("LogMacroPath", "/tmp")
+                    self.logging_path = macro_obj.getEnv("LogMacroPath")
+                log_file = self.logging_path + "/spock_session.log"
+                self.macro_obj.fileHandler = logging.FileHandler(log_file)
+                formatter = logging.Formatter(
+                    "%(levelname)-8s %(asctime)s %(name)s: %(message)s")
```

Comment:
What do you think to allow the user to change the formatter? Maybe another environment variable?

Reply:
It is fine for me. I can leave this as default in case the environment variable is not defined.

Reply:
Ok for me.